util/misc.go

```diff
@@ -53,3 +57,46 @@ func GetStack() []byte {
 	buf = buf[:stackSize]
 	return buf
 }
+
+// WithRecovery wraps goroutine startup call with recovery handle.
+// There are two out of box common handlers `WithPanicLogf` and `WithPanicMetricInc`,
+// Caller can custom define new handler use last optional parameters.
+func WithRecovery(ctx context.Context, fn func(ctx context.Context), handlerOpts ...func(r interface{})) {
```

Comment:
By the way, I think the prototype of `fn` can be simplified to 
```go
fn func()
```

Let the caller to construct the lambda function.

Reply:
In this way, it seems the `ctx` parameter for function `WithRecovery` can also be saved. Thus the function signature can be:
```go
func WithRecovery(callee func(), recoverer func(r interface{})) {
}
```

Reply:
if trace no need do some special action, +1 to remove ctx. @zhexuany 

recoverer be a variable arguments's benefit is we predefine some common handler, just like log, metric then caller can combine some them directly

Reply:
how about predefining some common handlers, combine them in the function body of the `recoverer()`?

Reply:
caller need write more code, combine in params will be more DSL feeling, 

`withRecovery  doSth withLog withMetric withMoreAction`

:laughing:

Reply:
how about changing the function signature to:
```go
func WithRecovery(callee func(), onPanic func(r interface{}), onFinish func(ctx context.Context)) {
}
```

Reply:
that's `try-catch-finally` :laughing: 

@coocood  how do you think about it~

Reply:
It's a little complicated, let's think a simpler way.